.github/actions/tscommon/Makefile

```
@@ -0,0 +1,48 @@
+SHELL := /bin/bash
+ACTION_NAME = $(shell basename "$$(pwd)")
+export UNIT_TESTS_WD := $(shell echo "$$(pwd)")
+
+.PHONY: help
+help: ## Shows all targets and help from the Makefile (this message).
+	@echo "$(ACTION_NAME) Makefile"
+	@echo "Usage: make [COMMAND]"
+	@echo ""
+	@grep --no-filename -E '^([/a-z.A-Z0-9_%-]+:.*?|)##' $(MAKEFILE_LIST) | \
+		awk 'BEGIN {FS = "(:.*?|)## ?"}; { \
+			if (length($$1) > 0) { \
+				printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2; \
+			} else { \
+				if (length($$2) > 0) { \
+					printf "%s\n", $$2; \
+				} \
+			} \
+		}'
+
+node_modules/.installed: package.json package-lock.json
+	npm ci
+	touch node_modules/.installed
+
+.PHONY: action
+action: node_modules/.installed ## Builds the action.
+	npm run build
+
+.PHONY: package
+package: node_modules/.installed ## Builds the distribution package.
+	npm run all
+
+.PHONY: clean
+clean:
+	rm -rf dist lib node_modules
+
+## Testing
+#####################################################################
+
+.PHONY: unit-test
+unit-test: node_modules/.installed ## Runs all unit tests.
+	# NOTE: Make sure the package builds.
+	npm run build
+	npm run test
+
+.PHONY: lint
+lint: node_modules/.installed ## Runs eslint.
+	npm run lint
```

nit: add `format-check` to make sure the code is formatted.
```suggestion
        npm run format-check
	npm run lint
```

---

done.